R/db.R

```diff
@@ -55,12 +55,92 @@ sql_join_suffix.DBIConnection <- function(con, suffix, ...) {
 
 #' @rdname db-misc
 #' @export
-db_sql_render <- function(con, sql, ..., cte = FALSE, use_star = TRUE) {
+db_sql_render <- function(con, sql, ..., cte = FALSE, sql_options = NULL) {
+  check_bool(cte)
+  if (cte) {
+    lifecycle::deprecate_soft(
+      when = "2.4.0",
+      what = "db_sql_render(cte)",
+      with = I("db_sql_render(sql_options = dbplyr_sql_options(cte = TRUE))")
+    )
+    sql_options <- dbplyr_sql_options(cte = TRUE)
+  }
+
+  if (is.null(sql_options)) {
+    sql_options <- dbplyr_sql_options()
+
+    out <- db_sql_render(con, sql, sql_options = sql_options)
```

Comment:
I almost wonder if it's better to do something like:

```R
db_sql_render <- function(con, sql, ..., cte = FALSE, sql_options = NULL) {
   ...
   db_sql_render2(con, sql, ..., sql_options = sql_options)
}

db_sql_render2 <- function(con, sql, ..., cte = FALSE, sql_options = NULL) {
  UseMethod("db_sql_render")
}
```

I think we also need either a `as_sql_options()` or `check_sql_options()` to ensure that `sql_option` is actually an list of the correct type.